crates/next-api/src/project.rs

```diff
@@ -173,25 +173,46 @@ pub struct Instrumentation {
 
 #[turbo_tasks::value]
 pub struct ProjectContainer {
-    options_state: State<ProjectOptions>,
+    name: RcStr,
+    options_state: State<Option<ProjectOptions>>,
     versioned_content_map: Option<Vc<VersionedContentMap>>,
 }
 
 #[turbo_tasks::value_impl]
 impl ProjectContainer {
     #[turbo_tasks::function]
-    pub fn new(options: ProjectOptions) -> Vc<Self> {
+    pub fn new(name: RcStr, dev: bool) -> Vc<Self> {
         ProjectContainer {
+            name,
             // we only need to enable versioning in dev mode, since build
             // is assumed to be operating over a static snapshot
-            versioned_content_map: options.dev.then(VersionedContentMap::new),
-            options_state: State::new(options),
+            versioned_content_map: dev.then(VersionedContentMap::new),
+            options_state: State::new(None),
         }
         .cell()
     }
+}
 
-    #[turbo_tasks::function]
-    pub fn update(&self, options: PartialProjectOptions) -> Vc<()> {
+impl ProjectContainer {
+    #[tracing::instrument(level = "info", name = "initialize project", skip_all)]
+    pub async fn initialize(self: Vc<Self>, options: ProjectOptions) -> Result<()> {
```

Comment:
Some sort of comment here (or on `new`) explaining what you explained in the diff summary would be very useful. Otherwise it's unintuitive why this `new`/`initialize` split exists, and I'd see myself trying to refactor this to group them back together.